core/frontend/src/main/scala/io/udash/core/Definitions.scala

```diff
@@ -94,19 +96,20 @@ trait ContainerView extends View {
 trait FinalView extends View
 
 /** The class which should be used to present the state for [[io.udash.routing.RoutingEngine]]. */
-trait State[HierarchyRoot <: State[HierarchyRoot]] {
-  def parentState: Option[ContainerState[HierarchyRoot] with HierarchyRoot]
+trait State {
+  type HierarchyRoot <: State { type HierarchyRoot = State.this.HierarchyRoot }
```

Comment:
I don't know why, but Scalatest sometimes thinks that broken code is compiling... Anyway take a look at `StateTest` class.